app/assets/javascripts/discourse/app/components/discovery/filter-navigation.gjs

```diff
@@ -81,6 +124,462 @@ export default class DiscoveryFilterNavigation extends Component {
+    const dateOptions = [
+      { value: "1", key: "yesterday" },
+      { value: "7", key: "last_week" },
+      { value: "30", key: "last_month" },
+      { value: "365", key: "last_year" },
+    ];
+
+    return dateOptions
+      .filter((option) => {
+        const description = i18n(`filter.description.days.${option.key}`);
+        return (
+          !lastTerm ||
+          option.value.includes(lastTerm) ||
+          description.toLowerCase().includes(lastTerm.toLowerCase())
+        );
+      })
+      .map((option) => ({
+        name: `${prefix}${filterName}:${prevTerms}${option.value}`,
+        description: i18n(`filter.description.${option.key}`),
+        isPlaceholderCompletion: true,
+        term: option.value,
+      }));
+  }
+
+  getNumberSuggestions(prefix, filterName, prevTerms, lastTerm) {
+    const numberOptions = [
+      { value: "0" },
+      { value: "1" },
+      { value: "5" },
+      { value: "10" },
+      { value: "20" },
+    ];
+
+    return numberOptions
+      .filter((option) => !lastTerm || option.value.includes(lastTerm))
+      .map((option) => ({
+        name: `${prefix}${filterName}:${prevTerms}${option.value}`,
+        isPlaceholderCompletion: true,
+        term: option.value,
+      }));
+  }
+
+  /* selecting an item ------------------------------------------------------- */
+  @action
+  selectItem(item) {
+    const words = this.currentInputValue.split(/\s+/);
+
+    if (item.isPlaceholderCompletion) {
+      words[words.length - 1] = item.name;
+      let updatedValue = words.join(" ");
+      if (
+        !updatedValue.endsWith(":") &&
+        (!item.delimiters || item.delimiters.length < 2)
+      ) {
+        updatedValue += " ";
+      }
+      this.updateValue(updatedValue);
+      this.searchResults = [];
+      this.updateResults(updatedValue);
+    } else {
+      const lastWord = words.at(-1);
+      const prefix = this.extractPrefix(lastWord);
+      const supportsPrefix = item.prefixes && item.prefixes.length > 0;
+      const filterName =
+        supportsPrefix && prefix ? `${prefix}${item.name}` : item.name;
+
+      words[words.length - 1] = filterName;
+      if (!filterName.endsWith(":") && !item.delimiters?.length) {
+        words[words.length - 1] += " ";
+      }
+
+      const updatedValue = words.join(" ");
+      this.updateValue(updatedValue);
+
+      const baseFilterName = item.name.replace(/^[-=]/, "").split(":")[0];
+      if (item.type) {
+        this.activeFilter = baseFilterName;
+        this.handlePlaceholderSearch(baseFilterName, "", item, prefix);
+      }
+    }
+
+    this.selectedIndex = -1;
+
+    next(() => {
+      this.inputElement.focus();
+      this.inputElement.setSelectionRange(
+        this.currentInputValue.length,
+        this.currentInputValue.length
+      );
+      this.updateResults(this.currentInputValue);
+    });
+  }
+
+  updateValue(value) {
+    this.newQueryString = value;
+    this.args.updateQueryString(value);
```

Comment:
name should be something like `onChange`, what we do with the new value is implementation detail of the parent component